### PR TITLE
es6: generators

### PR DESCRIPTION
**Pull request by ikokostya**
Relevant to https://github.com/ymaps/codestyle/issues/33

### CODE REVIEW

**Comment by dmikis** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -840,6 +841,75 @@ util.inherits(Stream, EventEmitter);
 
 [⬆ back to TOC](#table-of-contents)
 
+## Generators
+
+* Asterisk `*` in a [generator declaration](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*)
```
Comment (dodev) [inline README.md]: The asterisk

File: README.md [hunk 2]
```diff
@@ -840,6 +841,75 @@ util.inherits(Stream, EventEmitter);
...
+function *createIterator() {
+    yield 1;
+}
+
+const createIterator = function * () {
+    yield 1;
+};
+```
+
+* In [shorthand method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions)
```
Comment (dodev) [inline README.md]: In a shorthand... the asterisk

File: README.md [hunk 3]
```diff
@@ -840,6 +841,75 @@ util.inherits(Stream, EventEmitter);
...
+
+**Bad:**
+
+```js
+class Graph {
+    * edges() {}
+}
+```
+
+* In [yield* expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield*)
```
Comment (dodev) [inline README.md]: In an yield* ... the asterisk

**Comment by dodev** [comment]
:+1:

**Pull request merged**